### PR TITLE
Remove a buttload of junk

### PR DESCRIPTION
**Pull request by jbrockmendel**
Including one function literally named "junk"

Drops about 1.3% of the code base without breaking a sweat.

@kshedden I got the impression you're more-than-averagely amenable to getting rid of unused, unmaintained, would-raise-NameError-if-ever-run code.

### CODE REVIEW

**Comment by josef-pkt** [comment]
This pull request **introduces 10 alerts** and **fixes 365** when merging d8e071cac8c7bb77f8516178ea0b22297eaf41de into f5b0ee6a9cfb50623ca962a3ce8a620cc21eac38 - [view on lgtm.com](https://lgtm.com/projects/g/statsmodels/statsmodels/rev/pr-1c49f9b47938ba315583f429cad7b7a93c31eaaf)

**new alerts:**

* 9 for Unreachable code
* 1 for Variable defined multiple times

**fixed alerts:**

* 338 for Module\-level cyclic import
* 23 for Variable defined multiple times
* 2 for Potentially uninitialized local variable
* 1 for Wrong name for an argument in a call
* 1 for Unreachable code

---

*Comment posted by [lgtm.com](https://lgtm.com)*

**Comment by josef-pkt** [comment]
you think I'm going to merge this?

fails on all ci testing, and includes
```
+            raise NotImplementedError("`power` and `alpha` variables are "
+                                      "not defined.  This section of "
+                                      "statsmodels is not maintained.")
```

**Comment by jbrockmendel** [comment]
Of course not, this is illustrative of low-hanging fruit for trimming.

Those NotImplementedErrors raise instead of NameErrors that would get raised if that code were ever hit. This way it’s at least explicit/intentional.

File: statsmodels/stats/power.py [hunk 1]
```diff
@@ -223,6 +223,9 @@ def solve_power(self, **kwds):
             import warnings
             from statsmodels.tools.sm_exceptions import HypothesisTestWarning
             warnings.warn('Warning: Effect size of 0 detected', HypothesisTestWarning)
+            raise NotImplementedError("`power` and `alpha` variables are "
+                                      "not defined.  This section of "
+                                      "statsmodels is not maintained.")
```
Comment (josef-pkt) [inline statsmodels/stats/power.py]: AFAICS or AFAIR, the NotImplementedError is just wrong her.

The code should work as expected. If effect size is zero, then null and alternative hypothesis are the same and power == alpha.

**Comment by josef-pkt** [comment]
I spot checked two cases. They break the code or some applications and are definitely NOT cleanup changes.

low-hanging fruits can be pretty sour.

Everything has at least minimal maintenance.
Specifically, the `power` case was only changed a few months ago https://github.com/statsmodels/statsmodels/pull/4087/files

(Even if there were no maintenance, we wouldn't announce it in an exception message.)

File: statsmodels/stats/outliers_influence.py [hunk 2]
```diff
@@ -665,79 +665,6 @@ def summary_table(self, float_fmt="%6.3f"):
 
 
 def summary_table(res, alpha=0.05):
```
[the remark this answers is not in the document]
Comment (jbrockmendel) [reply statsmodels/stats/outliers_influence.py]: Fair enough, will revert.  Its still mightily redundant with OLSInfluence.summary_table.

**Comment by josef-pkt** [comment]
This pull request **introduces 9 alerts** and **fixes 365** when merging ecbbf0324d595ba674fefe9b157db0a345672a14 into f5b0ee6a9cfb50623ca962a3ce8a620cc21eac38 - [view on lgtm.com](https://lgtm.com/projects/g/statsmodels/statsmodels/rev/pr-f655877d875b2e9b6a1ce67e3f5a37e885bb1335)

**new alerts:**

* 8 for Unreachable code
* 1 for Variable defined multiple times

**fixed alerts:**

* 338 for Module\-level cyclic import
* 23 for Variable defined multiple times
* 2 for Potentially uninitialized local variable
* 1 for Wrong name for an argument in a call
* 1 for Unreachable code

---

*Comment posted by [lgtm.com](https://lgtm.com)*

**Comment by jbrockmendel** [comment]
Closing to shift focus to many “real” PRs.

**Pull request closed without merging**